pennylane/tape/tape.py

```diff
@@ -435,6 +439,7 @@ def __exit__(self, exception_type, exception_value, traceback):
         QueuingManager.remove_active_queue()
         QuantumTape._lock.release()
         self._process_queue()
+        self._trainable_params = None
```

Comment:
why do we need this?

Reply:
One of the tests was breaking without it. I think because `_process_queue` calls `_update`, and `_update` no longer reset the trainable params.